staking/integration-tests/tests/stability.rs

```diff
@@ -0,0 +1,439 @@
+            delegator: usize::arbitrary(g) % delegators.len(),
+        }
+    } else if rnd
+        < delegate_weight + undelegate_weight + advance_delegation_record_weight + advance_weight
+    {
+        Operation::Advance {
+            cap: u64::arbitrary(g) % MAX_CAP + 1,
+        }
+    } else {
+        Operation::Slash {
+            publisher: usize::arbitrary(g) % publishers.len(),
+            amount:    u64::arbitrary(g) % (FRAC_64_MULTIPLIER - 1) + 1,
+        }
+    }
+}
+
+fn handle_delegate_operation(
+    svm: &mut LiteSVM,
+    pool_data_pubkey: Pubkey,
+    delegator: &Keypair,
+    publisher: Pubkey,
+    amount: u64,
+    stake_account_positions: Pubkey,
+) {
+    let stake_account_custody = get_stake_account_custody_address(stake_account_positions);
+
+    let custody_data: TokenAccount = fetch_account_data(svm, &stake_account_custody);
+
+    let mut stake_positions_account = fetch_positions_account(svm, &stake_account_positions);
+    let positions = stake_positions_account.to_dynamic_position_array();
+
+    let mut total_delegate = 0;
+
+    for i in 0..positions.get_position_capacity() {
+        if let Some(position) = positions.read_position(i).unwrap() {
+            if position.target_with_parameters.get_target() == Target::IntegrityPool {
+                total_delegate += position.amount;
+            }
+        }
+    }
+
+    let res = delegate(
+        svm,
+        delegator,
+        publisher,
+        pool_data_pubkey,
+        stake_account_positions,
+        amount,
+    );
+
+    if total_delegate + amount > custody_data.amount {
+        assert_anchor_program_error!(res, StakingErrorCode::TooMuchExposureToIntegrityPool, 0);
+    } else {
+        res.unwrap();
+    }
+}
+
+fn test_stability(props: StabilityTestProps) {
+    let SetupResult {
+        mut svm,
+        payer,
+        pyth_token_mint,
+        publisher_keypair: _,
+        pool_data_pubkey,
+        reward_program_authority,
+        publisher_index: _,
+    } = setup(SetupProps {
+        init_config:     true,
+        init_target:     true,
+        init_mint:       true,
+        init_pool_data:  true,
+        init_publishers: false,
+    });
+
+    props.delegators.iter().for_each(|delegator| {
+        svm.airdrop(&delegator.0.pubkey(), 1_000_000_000).unwrap();
+    });
+
+    let stake_account_positions: Vec<Pubkey> = props
+        .delegators
+        .iter()
+        .map(|delegator| {
+            initialize_new_stake_account(&mut svm, &delegator.0, &pyth_token_mint, true, true)
+        })
+        .collect();
+
+    initialize_pool_reward_custody(&mut svm, &payer, &pyth_token_mint);
+
+    let publisher_caps = post_publisher_caps(
+        &mut svm,
+        &payer,
+        props.publishers.iter().map(|p| p.0.pubkey()).collect(),
+        vec![0; props.publishers.len()],
+    );
+    advance(&mut svm, &payer, publisher_caps).unwrap();
+
+    for operation in props.operations.iter() {
```

Comment:
feels like every N operations you should check pool accounting by scanning the delegators positions and summing them up then comparing against pool data